File: Python/project-euler/problems/problem_007.py
from utils.code_runner import execute_code


def is_prime(number):
    if number <= 1:
        return False
    if number == 2:
        return True
    if number % 2 == 0:
        return False

    counter = 3

    while counter * counter <= number:
        if number % counter == 0:
            return False

        counter += 2

    return True


def problem():
    target = 10001

    number_of_primes, number = 1, 1

    while number_of_primes < target:
        number += 2
        if is_prime(number):
            number_of_primes += 1

    return number


if __name__ == '__main__':
    execute_code(problem)

File: Java/src/main/java/nl/davydehaas/projecteuler/Problem007.java
package nl.davydehaas.projecteuler;

import static nl.davydehaas.projecteuler.util.SolutionTimer.timeSolution;

public class Problem007 {

    void main() {
        timeSolution(Problem007::solve);
    }

    static Number solve() {
        int target = 10001;
        int numberOfPrimes = 1;
        int number = 1;

        while (numberOfPrimes < target) {
            number += 2;
            if (isPrime(number)) {
                numberOfPrimes++;
            }
        }
        // The 10001th prime number is
        return number;
    }

    static boolean isPrime(int number) {
        if (number < 2) {
            return false;
        }
        if (number == 2) {
            return true;
        }
        if (number % 2 == 0) {
            return false;
        }

        int counter = 3;
        while (counter * counter <= number) {
            if (number % counter == 0) {
                return false;
            }
            counter += 2;
        }
        return true;
    }
}
